capstone/src/test/java/com/google/step/clients/MetricsClientTests.java

```diff
@@ -0,0 +1,353 @@
+
+    @Test
+    public void testPutPageView() {
+        // Checks if the putPageView method is correctly putting data into dataStore.
+        metricsClient.putPageView("Mc Donald's", "1");
+        Calendar calendar = Calendar.getInstance();
+        int year = calendar.get(Calendar.YEAR);
+        int week = calendar.get(Calendar.WEEK_OF_YEAR);
+
+        CompositeFilter compositeFilter = new Query.CompositeFilter(Query.CompositeFilterOperator.AND,
+                Arrays.asList(
+                        new FilterPredicate("restaurantKey", Query.FilterOperator.EQUAL, "1"),
+                        new FilterPredicate("year", Query.FilterOperator.EQUAL, year),
+                        new FilterPredicate("week", Query.FilterOperator.EQUAL, week)
+                ));
+
+        Query query = new Query("PageViews").setFilter(compositeFilter);
+        PreparedQuery results = datastoreService.prepare(query);
+        Entity resultEntity = results.asSingleEntity();
+        Assert.assertEquals("Count", (long) 1, resultEntity.getProperty("count"));
+    }
+
+    @Test
+    public void testSeveralPutPageViews() {
+        // Checks if the putPageView method is correctly putting and updating an entity in dataStore.
+        metricsClient.putPageView("Mc Donald's", "1");
+        metricsClient.putPageView("Mc Donald's", "1");
+        metricsClient.putPageView("Mc Donald's", "1");
+        Calendar calendar = Calendar.getInstance();
+        int year = calendar.get(Calendar.YEAR);
+        int week = calendar.get(Calendar.WEEK_OF_YEAR);
+
+        CompositeFilter compositeFilter = new Query.CompositeFilter(Query.CompositeFilterOperator.AND,
+                Arrays.asList(
+                        new FilterPredicate("restaurantKey", Query.FilterOperator.EQUAL, "1"),
+                        new FilterPredicate("year", Query.FilterOperator.EQUAL, year),
+                        new FilterPredicate("week", Query.FilterOperator.EQUAL, week)
+                ));
+
+        Query query = new Query("PageViews").setFilter(compositeFilter);
+        PreparedQuery results = datastoreService.prepare(query);
+        Entity resultEntity = results.asSingleEntity();
+        Assert.assertEquals("Count", (long) 3, resultEntity.getProperty("count"));
+    }
+
+    @Test
+    public void testGetYearRestaurantPageViews() {
+        // Tests if method is correctly grabbing the correct data for a specified year and restaurant.
+        // Result: [
+        //              WeeklyPageView with the following data:.
+        //                  (
+        //                      "week": 2,
+        //                      "year": 2015,
+        //                      "count": 100
+        //                  ),
+        //              WeeklyPageView with the following data.
+        //                  (
+        //                      "week": 2,
+        //                      "year": 2015,
+        //                      "count": 100
+        //                  )
```

Comment:
These look the same object. Are they supposed to have different values?

Reply:
Yes they are meant to have different values. Fixed.